src/server/jobs/update-videores.js

```diff
@@ -0,0 +1,130 @@
+const FtpClient = require('ftp');
+const winston = require('winston');
+require('dotenv').config();
+const { MongoClient } = require('mongodb');
+const RegistrarService = require('../services/registrar');
+const UpdateVideoResServices = require('../services/UpdateVideoResServices');
+
+// Global variable for current date and time
+const currentTimestamp = new Date();
+
+// Global logger that outputs log messages to console and update-videores_YEAR-MONTH-DAY_HOUR-MIN.log
+// The file receives any log messages designated at least at 'info' level or higher priority
+const logger = winston.createLogger({
+  level: 'info',
+  format: winston.format.combine(
+    winston.format.timestamp({
+      format: 'YYYY-MM-DD HH:mm:ss',
+    }),
+    winston.format.json(),
+    winston.format.prettyPrint()
+  ),
+  transports: [
+    new winston.transports.Console(),
+    new winston.transports.File({
+      filename: `logs/update-videores_${currentTimestamp.getFullYear()}-${currentTimestamp.getMonth() +
+        1}-${currentTimestamp.getDate()}_${currentTimestamp.getHours()}-${currentTimestamp.getMinutes()}.log`,
+      level: 'debug',
+    }),
+  ],
+});
+
+/**
+ * The main process
+ */
+async function main() {
+  const dbURL = `${process.env.DB_URL}${process.env.DB_DATABASE}?replicaSet=${process.env.DB_REPLSET}`;
+  const dbclient = new MongoClient(dbURL, { useUnifiedTopology: true });
+  await dbclient.connect();
+  logger.info(`Connected database at ${dbURL}`);
+
+  let result = [];
+  const srsShortnameMap = [];
+
+  const c = new FtpClient();
+  c.on('ready', function() {
+    c.get(process.env.VIDEORES_FILEPATH, function(err, stream) {
+      if (err) throw err;
+      stream.once('close', function() {
+        c.end();
+      });
+      let str = '';
+      stream.on('data', data => {
+        str += data.toString();
+      });
+      stream.on('end', async () => {
+        result = UpdateVideoResServices.readStrIntoFields(str, logger);
+      });
+    });
+  });
+  c.once('end', async function() {
+    // Create an array of courses and get shortname for each course
+    for (const media of result) {
+      const mappedPair = srsShortnameMap.filter(
+        pair => pair.srs === media.srs && pair.term === media.term
+      );
+      if (mappedPair.length !== 0) {
+        if (!media.filename.includes(mappedPair[0].classShortname)) {
+          logger.warn(
+            `File ${media.filename} does not include class shortname ${mappedPair[0].classShortname}`
```

Comment:
@maxpluez When reviewing errors it isn't helpful to know the filename doesn't map to a shortname. We want to know what the term-srs is that doesn't map to a file name. And we only want to see that warning message once for a given term-srs pairing